### DISCUSSION TITLE
Create a resource with a default status ?

### DISCUSSION THREAD

**Discussion by EliXorZz** [kube-rs/kube#961] [Q&A] [answered]
Hello, I have created my **own resource definition** and I would like to be able to assign it a **default status** when creating this resource.
I found this in the examples [here](https://github.com/kube-rs/kube-rs/blob/e84cca0ec1baefb7659c5249cb7f752a831833ba/examples/crd_api.rs#L123-L130)

However, the status field is absolutely not taken into account during creation, is this normal?
I may have forgotten a parameter that I do not know?

**Thanks in advance for those who will help me!**

**Comment by clux** [comment, marked as answer]
Yeah, this is normal. Applies (via `patch`, `create` or `replace`) only affect the main part of the resource (i.e. the spec part). The same is true for `kubectl apply`.

Status objects are designed to only be set by controllers and need special methods to be set (`patch_status`, `create_status`, `replace_status`). I have never seen a resource with a default status (unless you count null/None).

**Reply by EliXorZz** [reply]
Yes, that's what I thought, but you should update your example, right?  Because it is false.

**Reply by clux** [reply]
Ah, yes you are right, thank you!. Fixed in https://github.com/kube-rs/kube-rs/pull/963
